lib/cylc/task_events_mgr.py

```diff
@@ -819,7 +830,11 @@ def _setup_custom_event_handlers(self, itask, event, message):
             retry_delays = [0]
         # There can be multiple custom event handlers
         for i, handler in enumerate(handlers):
-            key1 = ("%s-%02d" % (self.HANDLER_CUSTOM, i), event)
+            if is_generic:
+                key1 = ('%s-%02d' % (self.HANDLER_CUSTOM, i),
+                        '%s-%s' % (event, urllib.quote(message)))
+            else:
+                key1 = ('%s-%02d' % (self.HANDLER_CUSTOM, i), event)
```

Comment:
Is this distinction necessary?  If we keyed all events on message the code would be a bit simpler, and it wouldn't actually matter that message isn't needed to uniquely identify non-generic events, would it?

Reply:
You are right here. I'll investigate the effect of having a single `key1` on our current test battery and on restart, etc.

Reply:
(holding off on the till you report back on this...)

Reply:
The latest logic is a bit different. Key points are:
* For unique events, use event as key1. Easy for the eyes (when inspecting the logs) and no change required for tests.
* For non-unique events, use event + count(event) as key1.

Restart looks safe either way.

Reply:
OK, seems sensible.